oauthproxy.go

```diff
@@ -934,6 +934,12 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 		return nil, ErrNeedsLogin
 	}
 
+	// Make sure groups and email data is set, even if we believe it (should) already exist here
+	err := p.provider.EnrichSession(req.Context(), session)
```

Comment:
In many providers `EnrichSession` makes 1 or more network calls to an IdP ProfileURL.

So we only want it called sparingly at the end of the initial login flow. Being called here would results in extra network requests on every request I believe.